### ISSUE TITLE
Improve sound control

### ISSUE THREAD

**Issue by khromov** [replit/kaboom#33]
👋 I'm making a proof of concept of a rhythm game using Kaboom. It's such a simple and easy-to-understand engine & it's great to work in.

It would be very useful to have additional control of the `play()` function, mainly:

1. Being able to ask the underlying player how long it has been playing:

```js
const music = play("song1");
//...
music.currentTime();
```

2. Being able to attach event listeners to the audio `timeupdate` event and other events from the underlying audio player so you don't have to poll for updates, similar to `obj.on()`

```js
const music = play("song1");
//...
music.on("timeupdate", () => {
    //...
});

**Comment by slmjkdbtl** [comment]
Looks like these are 2 features by the DOM media elements, the `currentTime()` could be trivial to implement, but I'm not too sure on what's the mechanism of `"timeupdate"` being fired, on DOM media elements `currentTime` seem do seem update in between `"timeupdate"` events

**Comment by khromov** [comment]
@slmjkdbtl Exactly, these are two features of `HTMLMediaElement`.  `timeupdate` triggers very frequently but depends a bit on system load, if the tab is idle and other factors. There is a good explanation in [the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/timeupdate_event).

I see that you are using `AudioBufferSourceNode`. There is some discussion [on StackOverflow](https://stackoverflow.com/questions/31644060/how-can-i-get-an-audiobuffersourcenodes-current-time) around how to handle

I think having `currentTime()` would be sufficient, since you can just poll the time using `obj.on("update" ...` from another object!

**Comment by slmjkdbtl** [comment]
Added in b1ce721 and this [example](https://kaboomjs.com/examples#audio)!

**Comment by khromov** [comment]
@slmjkdbtl Looks great! 
In the audio example you linked, pausing with spacebar doesn't seem to work for me on Firefox 88. It works in Chrome.

**Comment by slmjkdbtl** [comment]
Hmm so Firefox doesn't like setting `playbackRate` to `0` as a hacky audio pause, thanks for letting me know, I'll try something else for pausing!

**Comment by slmjkdbtl** [comment]
Fixed Firefox pausing issue in 1fe8d0a. Closing this for now, feel free to reopen if there are other audio related feature requests or discussions!
